src/bundle/mutable.js

```diff
@@ -0,0 +1,144 @@
+// Adapted from SES/Caja
+// Copyright (C) 2011 Google Inc.
+// https://github.com/google/caja/blob/master/src/com/google/caja/ses/startSES.js
+// https://github.com/google/caja/blob/master/src/com/google/caja/ses/repairES5.js
+
+export default function makeRepairDataProperties() {
+  const {
+    defineProperties,
+    getOwnPropertyDescriptors,
+    hasOwnProperty,
+  } = Object;
+  const { ownKeys } = Reflect;
+
+  // Object.defineProperty is allowed to fail silently,
+  // wrap Object.defineProperties instead.
+  function defineProperty(obj, prop, desc) {
+    defineProperties(obj, { [prop]: desc });
+  }
+
+  /**
+   * For a special set of properties (defined below), it ensures that the
+   * effect of freezing does not suppress the ability to override these
+   * properties on derived objects by simple assignment.
+   *
+   * Because of lack of sufficient foresight at the time, ES5 unfortunately
+   * specified that a simple assignment to a non-existent property must fail if
+   * it would override a non-writable data property of the same name. (In
+   * retrospect, this was a mistake, but it is now too late and we must live
+   * with the consequences.) As a result, simply freezing an object to make it
+   * tamper proof has the unfortunate side effect of breaking previously correct
+   * code that is considered to have followed JS best practices, if this
+   * previous code used assignment to override.
+   *
+   * To work around this mistake, deepFreeze(), prior to freezing, replaces
+   * selected configurable own data properties with accessor properties which
+   * simulate what we should have specified -- that assignments to derived
+   * objects succeed if otherwise possible.
+   */
+  function beMutable(obj, prop, desc) {
```

Comment:
It is probably my own fault, but this file should not be named "mutable.js" and this function should not be named `beMutable`. Neither reflects their actual meaning. Somewhere there's a more recent version of this repair code that improved some minor things, including, I think, these names.

Attn @caridy @jdalton @jfparadis What does the current Saleforce code say? Can you post it or point at it?

Attn @guybedford @bmeck What does the current Node code say? Can you post it or point at it?

Reply:
renamed file to `makeRepairDataProperties` and renamed internal `beMutable` fn to `enableDerivedOverride`